src/lib/reflected-attribute-name.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Elements and their special properties that should be reflected to attributes
+ * when set.
+ * 
+ * Modify this object to add reflected attributes.
+ *
+ * Each item in the array takes the following format:
+ *
+ * first: Element name or array of element names.
+ * rest:  Array of ([property name, reflected attribute name] or
+ *        property name (if reflected attribute name is identical).
+ */
+const reflectedAttributesSource: Array<{
+  tagNames: string[];
+  reflections: (string | string[])[];
+}> = [
+  {
+    tagNames: ['input', 'option'],
+    reflections: ['value']
+  },
+  {
+    tagNames: ['*'],
+    reflections: [['className', 'class'], 'id']
+  }
+];
+
+// reflectedAttributesSource is easy to visually parse, but we can reconfigure the
+// data structure into a map of maps for faster lookup at runtime.
+const reflectedAttributes = new Map<string, Map<string, string>>();
+const addReflectionForElement = (elementName: string, propertyName: string, attributeName: string) => {
+  if (reflectedAttributes.has(elementName)) {
```

Comment:
A super nit, but I like to keep a very regular pattern for accessing Maps like this:

```js
let reflectedAttribute = reflectedAttributes.get(elementName);
if (reflectedAttribute === undefined) {
  reflectedAttribute = new Map();
  reflectedAttributes.set(elementName, reflectedAttribute);
}
reflectedAttribute.set(propertyName, attributeName);
```

That keeps things very simple so that there's 1) a map lookup 2) handle the missing case 3) present and missing cases and handled the same.

In performance sensitive code it also eliminates another hash/lookup in `reflectedAttributes`. This probably isn't perf-sensitive, but always doing it this way means you don't have to think about it.